### ISSUE TITLE
False positives when using WPDB from a class property (e.g. `$this->wpdb`)

### ISSUE THREAD

**Issue by hannahtinkler** [WordPress/WordPress-Coding-Standards#1991]
## Bug Description
To help with testing and testability (especially mocking), I'm injecting the global `$wpdb` into a class constructor and using it in methods with `$this->`. This fails linting with `WordPress.DB.PreparedSQL.NotPrepared`.

Example code:
```
$this->wpdb->get_row(
	$this->wpdb->prepare(
		'SELECT * FROM mobile_push_tokens WHERE id = %d',
		$id
	)
);
```

Failures:
```
FOUND 4 ERRORS AFFECTING 2 LINES
-----------------------------------------------------------------------------------------------------------
 102 | ERROR | Use placeholders and $wpdb->prepare(); found $this
     |       | (WordPress.DB.PreparedSQL.NotPrepared)
 102 | ERROR | Use placeholders and $wpdb->prepare(); found wpdb
     |       | (WordPress.DB.PreparedSQL.NotPrepared)
 102 | ERROR | Use placeholders and $wpdb->prepare(); found prepare
     |       | (WordPress.DB.PreparedSQL.NotPrepared)
 104 | ERROR | Use placeholders and $wpdb->prepare(); found $id (WordPress.DB.PreparedSQL.NotPrepared)
```

Ideally the sniff would allow for usage of `$this->wpdb` as well as the usual `$wpdb`.

## Error Code
`WordPress.DB.PreparedSQL.NotPrepared`

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.3.28
| PHP_CodeSniffer version | 3.6.0
| WPCS version            | 2.3.0
| WPCS install type       | Composer project local
| IDE (if relevant)       | Using in iTerm, not an IDE

## Tested Against `develop` branch?
- [x] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
This is a known issue without a solution at this time.

Closing as duplicate of #1589
